src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -45,6 +45,9 @@ const (
 	configDestinationDir = "/etc/redpanda"
 	configSourceDir      = "/mnt/operator"
 	configFile           = "redpanda.yaml"
+
+	datadirName            = "datadir"
+	defaultDatadirCapacity = "100Gi"
```

Comment:
I know this is absolutely outside of scope of this PR but how will we handle resizing?

Reply:
also is 100Gi a sane default for redpanda? (I don't know, just making sure)

Reply:
Ability to dynamically resize depends on the storage class. If the storage class' (dynamic) provisioner supports it, the capacity change should go through. For example, persistent disks on GCE support expansion and so does its provisioner. Snippet of storage class:

```
provisioner: kubernetes.io/gce-pd
allowVolumeExpansion: true
```

Local volumes aren't covered by a dynamic provisioner. The local-static-provisioner (https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner), which we could use in the future also doesn't support it as far as I can tell. 

Having said that, if the underlying device was manually expanded (along with the filesystem), the capacity would become available. The PV (and PVC) capacity fields would not reflect that unless manually updated.

In practice, if one uses a local device of 1TB I'd expect they create a PV that specifies 1TB (similarly to the RFC example), so there's no possibility of resizing. When creating the CR, they should provide a capacity of 1TB (or less) such that the binding is successful. I should probably add a section in the RFC around resizing.

Reply:
+1 for the required field, that could be another validation in the webhook. Either here or create an issue, I am fine with either.

Reply:
@dimitriscruz If we doesn't allow for resizing the capacity in our Cluster CR we should disallow this in webhook.

Reply:
Resizing (expanding) is applicable under some storage classes but not under other storage classes.. so I'm not sure we'd want to remove this capability just yet.